.github/workflows/ci-tests.yml

```diff
@@ -0,0 +1,65 @@
+name: CI Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  ci-tests:
+    env:
+      VIRTUAL_ENV: github
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version:
+        - python: "2.7"
+          tornado: ">=4,<5"
+        - python: "2.7"
+          tornado: ">=5,<6"
+        - python: "3.5"
+          tornado: ">=4,<5"
+        - python: "3.5"
+          tornado: ">=5,<6"
+        - python: "3.5"
+          tornado: ">=6"
+        - python: "3.6"
+          tornado: ">=4,<5"
+        - python: "3.6"
+          tornado: ">=5,<6"
+        - python: "3.6"
+          tornado: ">=6"
+        - python: "3.7"
+          tornado: ">=4,<5"
+        - python: "3.7"
+          tornado: ">=5,<6"
+        - python: "3.7"
+          tornado: ">=6"
+    name: ci-tests (py ${{ matrix.version.python }} tornado ${{ matrix.version.tornado }})
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.version.python }}
+
+    - name: Export TORNADO variable
+      run: |
+        echo "TORNADO=${{ matrix.version.tornado }}" >> $GITHUB_ENV
+
+    - name: Install pycurl dependencies
+      run: sudo apt-get install libgnutls28-dev libcurl4-openssl-dev libssl-dev
+
+    - name: Install dependencies
+      run: make bootstrap
```

Comment:
do you need to echo VIRTUALENV to GITHUB_ENV first?

Reply:
Since we are setting VIRTUALENV at job level in #L11-12  it will automatically percolate down to all the steps.